### PR TITLE
[front] - fix(conversations): render conversations with archived agents

### LINKED ISSUE #8898: Render conversation with deleted agent
We no longer render conversations with agent that were archived. This behaviour was recently introduced by #8892.

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

 This PR modifies `fetchWorkspaceAgentConfigurationsWithoutActions` to return either active versions if they exist, or otherwise the latest versions.

This was recently introduced by https://github.com/dust-tt/dust/pull/8892. The goal of #8892 was to fix #8888.

**Context:** we recently found out that:
```
return AgentConfiguration.findAll({
          where: {
            workspaceId: owner.id,
            ...(agentPrefix ? { name: { [Op.iLike]: `${agentPrefix}%` } } : {}),
            sId: agentsGetView.agentIds.filter((id) => !isGlobalAgentId(id)),
          },
          order: [["version", "DESC"]],
          ...(agentsGetView.allVersions ? {} : { limit: 1 }),
        });
      }
``` 
was causing a bunch of errors on the `/participants` endpoint as it was limiting to return only one agent configuration. Because of this the conversation participants were broken.

The initial fix was to retrieve all "active" agent configurations, which fixed the conversation participants issue. However, this led to not be able to retrieve conversation with an "archived" agent. 

This PR ensures that we either return the active agent configuration or its latest archived version.

## Risk

Low

## Deploy Plan

Deploy `front`

### CODE REVIEW

**Comment by spolu** [comment]
@JulesBelveze 

So that I understand correctly this was the limit: 1 change right? 

So the diffs we had so far are:

https://github.com/dust-tt/dust/pull/8888/files
https://github.com/dust-tt/dust/pull/8892/files

And now this one?

Let's step back and really think through the change to make sure we have a full understanding here. To do so please update your PR description with references to previous 2 PRs what they changes what they broke and why this is guaranteed to be the right end-state :pray:

File: front/lib/api/assistant/configuration.ts [hunk 1]
```diff
@@ -366,15 +366,35 @@ async function fetchWorkspaceAgentConfigurationsWithoutActions(
       });
     default:
       if (typeof agentsGetView === "object" && "agentIds" in agentsGetView) {
-        return AgentConfiguration.findAll({
+        const agents = await AgentConfiguration.findAll({
```
Comment (flvndvd) [inline front/lib/api/assistant/configuration.ts:369]: It seems a bit tough to not leverage the SQL query, no? Agent configurations can have hundreds or versions 🤔

Comment (flvndvd) [reply front/lib/api/assistant/configuration.ts:369]: LEt's see the SQL query once you have it. Maybe we can map it to Sequelize.

**Comment by spolu** [comment]
> Because of this the conversation participants were broken.

@JulesBelveze do we understand when it broke and how it broke? This query has been here for a long time no? Yet participants have been working in the past? What exactly broke them?

**Comment by JulesBelveze** [comment]
@spolu I'm not sure to follow how this change was working: https://github.com/dust-tt/dust/commit/99771e8a1b95c1bf08e19ef0122611b237e56aa9

**Comment by spolu** [comment]
You mean the limit: 1 was at odd with the move to sId array I presume. Seems plausibe that it was problematic cc @philipperolet ?

Good that we're getting down to a very clear picture. Feel like we now have reached full context, which is great \o/

**Comment by spolu** [comment]
@JulesBelveze if you can add @philipperolet as reviewer this would be great to make sure we have full context on getting the good final state :pray: 

Thanks for digging!

**Comment by philipperolet** [comment]
After looking at it I concur with @JulesBelveze the query didn't work. Apologies for missing that. 

It's strange that nobody noticed since it's been 2 months, but at the same time since it only impacts the fact that you don't see all the avatars on the top right of your screen in a conversation it's possible

Thanks a lot for solving this @JulesBelveze

**Review by philipperolet — COMMENTED** [review]
1. LGTM  and thanks on fixing the issue from my PR moving to agentIds array :pray: 
2. however, I think the 'status:active' filter is what's breaking the preview; AFAICT, the preview is grabbing for a 'draft' ? continuing investigation there

**Comment by philipperolet** [comment]
> 1. LGTM  and thanks on fixing the issue from my PR moving to agentIds array 🙏
> 2. however, I think the 'status:active' filter is what's breaking the preview; AFAICT, the preview is grabbing for a 'draft' ? continuing investigation there

For reference, I'm talking about this wrt the preview broken: https://github.com/dust-tt/tasks/issues/1703

**Comment by philipperolet** [comment]
Other topic: conversations with archived agents generate unhandled errors

=> https://dust4ai.slack.com/archives/C05F84CFP0E/p1732630658549379

**Comment by philipperolet** [comment]
Discussed IRL:
- @JulesBelveze proposed grouping by agentId and taking the latest version instead of status filtering
- this should solve 2 problems mentioned above, and avoid the raw SQL query
:pray:

**Review by philipperolet — APPROVED** [review]
:+1: (make sure to test)

**Pull request merged**